### ISSUE TITLE
findAllAsListAsync and listAllAsync don't work

### ISSUE THREAD

**Issue by takezoe** [bizreach/elastic-scala-httpclient#21]

**Comment by ninneko** [comment]
https://github.com/bizreach/elastic-scala-httpclient/blob/175851d1e75d74898b125fe76f54669efc643153/elastic-scala-httpclient/src/main/scala/jp/co/bizreach/elasticsearch4s/AsyncESClient.scala#L164

this method does not work.
listAsync need countAsIntAsync's result, so this method should be written as following.

```
  def listAllAsync[T](config: ESConfig)(f: SearchRequestBuilder => Unit)(implicit c: ClassTag[T]): Future[ESSearchResult[T]] = {
    for {
      count <- countAsIntAsync(config)(f)
      result <- listAsync[T](config) { searcher =>
        f(searcher)
        searcher.setFrom(0)
        searcher.setSize(count)
      }
    } yield result
  }
```

**Comment by shimamoto** [comment]
Oh, as you say, this method is considered not operating as intended. I will fix these bugs after releasing 3.2.4. Thank you for letting me know!

**Comment by shimamoto** [comment]
It's available in 3.2.5.
